gensim/summarization/mz_entropy.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Licensed under the GNU LGPL v2.1 - http://www.gnu.org/licenses/lgpl.html
+
+
+from gensim.summarization.textcleaner import tokenize_by_word as _tokenize_by_word
+from gensim.utils import to_unicode
+import numpy
+import scipy
+
+def mz_keywords(text,blocksize=1024,scores=False,split=False,weighted=True,threshold=0.0):
+    """Extract keywords from text using the Montemurro and Zanette entropy algorithm.
+       https://arxiv.org/abs/0907.1558
+       :param text: str (document to summarize)
+       :param blocksize: int (size of blocks to use in analysis)
+       :params scores: bool (return score with keywords)
+       :params split: bool (return results as list)
+       :params weighted: bool (weight scores by word frequency. 
+                               False is useful for shorter texts)
+       :params threshold: float or 'auto' (minimum score for returned keywords
+                                           'auto' calculates the threshold as 
+                                           nblocks/(nblocks+1.0)
+                                           Use 'auto' with weighted=False)"""
+    text=to_unicode(text)
+    words=[word for word in _tokenize_by_word(text)]
+    vocab=sorted(set(words))
+    wordcounts=numpy.array([[words[i:i+blocksize].count(word) for word in vocab]
+                            for i in range(0,len(words),blocksize)]).astype('d')
+    nblocks=wordcounts.shape[0]
+    totals=wordcounts.sum(axis=0)
+    nwords=totals.sum()
+    p=wordcounts/totals
+    logp=numpy.log2(p)
+    H=numpy.nan_to_num((p*logp),0.0).sum(axis=0)
+    
+    def log_combinations(n,m):
```

Comment:
Better move definition `log_combinations`, `marginal_prob`, `marginal` outside and start names with `_`.

Reply:
Refactoring the core functionality into a private class to address this issue

Reply:
yes, it's possible option